### DISCUSSION TITLE
Embeddable scripting language for V?

### DISCUSSION THREAD

**Discussion by BioBox** [vlang/v#17251] [Questions and Answers] [answered]
I'm thinking of making an embedded scripting language for V similar to what C is for Lua. The easiest way of doing this is likely to run `go2v` on [go-lua](https://github.com/Shopify/go-lua) and then put moonscript on it.

Is this a good idea? What do you all think?

**Comment by rraammiinn** [comment]
why not python using python.h headerfile ?

**Comment by BioBox** [comment]
Can you elaborate a little more? Has anyone been able to embed python in a V application? Also a lot of apps use Lua, so if I say, want to rewrite a C app in V, I'll have to redo the corresponding Lua API into a Python API as well.

**Reply by rraammiinn** [reply]
https://github.com/mvenditto/v-python
although it seems abandoned .

**Comment by JalonSolov** [comment]
There is a repo out showing how to do interop between Python and V, but nothing that tells how to "embed" Python in V.

Many people like lua for scripting, so if you can figure out how to do what you want, there's no reason not to try.

**Comment by medvednikov** [comment, marked as answer]
@BioBox V already has an interpreter mode, so it can be embedded.

It's at an early stage though.

vlib/v/eval

**Reply by rraammiinn** [reply]
can it support dynamic typing ?
some languages like c# can .

**Reply by JalonSolov** [reply]
No, the V interpreter is same as V itself - statically typed.

**Reply by rraammiinn** [reply]
but it has the potential of having any type like typescript as a scripting language .

**Reply by JalonSolov** [reply]
No, it doesn't, because that's not V.  All V code run in the V interpreter will run the same as it would if you compiled it.

**Reply by JalonSolov** [reply]
Yes, if the embedded language is something other than V.

**Reply by bogen85** [reply]
> but it has the potential of having any type like typescript as a scripting language .

Static typing is not a prerequisite nor is needed for a scripting or interpreted language.
V already has type inference and generics, it does not need dynamic typing.

One might prefer dynamic typing (and the abundance of problems it brings) over strong static typing, but that is a separate issue.

When you say "dynamic typing", in the context of an embedded scripting language it can be useful to reuse variable names. This could be done with independent (non global) interpreter states that and can created and destroyed on demand, and kept separate from each other.

**Reply by rraammiinn** [reply]
thanks

**Reply by rraammiinn** [reply]
I saw empty interface in go and I really didn't like it's syntax so I thought any keyword would be better than  {} .
